src/main/java/io/cdap/plugin/dynamo/source/DynamoDBBatchSourceConfig.java

```diff
@@ -0,0 +1,406 @@
+    "pipe sign '|' and specifies the tokens and its type. For example, ':v_id|int,:v_title|string', if the query is" +
+    " like: 'ID = :v_id AND Title = :v_title'. Supported types are: 'boolean, int, long, float, double and string'." +
+    " (Macro Enabled)")
+  @Macro
+  private String placeholderType;
+
+  @Name(READ_THROUGHPUT)
+  @Description("Read Throughput for AWS DynamoDB table to connect to, in double. Default is 1. (Macro Enabled)")
+  @Nullable
+  @Macro
+  private String readThroughput;
+
+  @Name(READ_THROUGHPUT_PERCENTAGE)
+  @Description("Read Throughput Percentage for AWS DynamoDB table to connect to. Default is 0.5. (Macro Enabled)")
+  @Nullable
+  @Macro
+  private String readThroughputPercentage;
+
+  @Name(SCHEMA)
+  @Description("Specifies the schema that has to be output. If not specified, then by default each item will be " +
+    "emitted as a JSON string, in the 'body' field of the StructuredRecord.")
+  private String schema;
+
+  public DynamoDBBatchSourceConfig(String referenceName, String accessKey, String secretAccessKey,
+                                   @Nullable String region, @Nullable String endpointUrl, String tableName,
+                                   String query, @Nullable String filterQuery, @Nullable String nameMappings,
+                                   String valueMappings, String placeholderType, @Nullable String readThroughput,
+                                   @Nullable String readThroughputPercentage, String schema) {
+    super(referenceName);
+    this.accessKey = accessKey;
+    this.secretAccessKey = secretAccessKey;
+    this.region = region;
+    this.endpointUrl = endpointUrl;
+    this.tableName = tableName;
+    this.query = query;
+    this.filterQuery = filterQuery;
+    this.nameMappings = nameMappings;
+    this.valueMappings = valueMappings;
+    this.placeholderType = placeholderType;
+    this.readThroughput = readThroughput;
+    this.readThroughputPercentage = readThroughputPercentage;
+    this.schema = schema;
+  }
+
+  private DynamoDBBatchSourceConfig(Builder builder) {
+    super(builder.referenceName);
+    accessKey = builder.accessKey;
+    secretAccessKey = builder.secretAccessKey;
+    region = builder.region;
+    endpointUrl = builder.endpointUrl;
+    tableName = builder.tableName;
+    query = builder.query;
+    filterQuery = builder.filterQuery;
+    nameMappings = builder.nameMappings;
+    valueMappings = builder.valueMappings;
+    placeholderType = builder.placeholderType;
+    readThroughput = builder.readThroughput;
+    readThroughputPercentage = builder.readThroughputPercentage;
+    schema = builder.schema;
+  }
+
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  public static Builder builder(DynamoDBBatchSourceConfig copy) {
+    Builder builder = new Builder();
+    builder.referenceName = copy.referenceName;
```

Comment:
it's better to assign these through the builder's setter methods rather than direct assignment, to ensure everything goes through the same code path.

Reply:
This code was generated by Intellij IDEA plugin. I thought it would be easier to make it similar in all classes and easier to update. Actually I usually prefer Lombok but tried to follow style from other plugins.
I will rewrite to address your comment.